modules/RockRuby.cmake

```diff
@@ -266,6 +266,11 @@ ELSEIF(NOT RUBY_EXTENSIONS_AVAILABLE)
        OUTPUT_VARIABLE RUBY_CFLAGS)
     STRING(REPLACE "\n" "" RUBY_CFLAGS ${RUBY_CFLAGS})
 
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        #Clang can't handle this option (04.2014) remove this flag therefore
+        STRING(REPLACE "-fstack-protector-strong" "" RUBY_CFLAGS ${RUBY_CFLAGS})
```

Comment:
On 24.03.2015 16:47, Sylvain Joyeux wrote:
>
> In modules/RockRuby.cmake 
> <https://github.com/rock-core/base-cmake/pull/12#discussion_r27040782>:
>
> > @@ -266,6 +266,11 @@ ELSEIF(NOT RUBY_EXTENSIONS_AVAILABLE)
> >         OUTPUT_VARIABLE RUBY_CFLAGS)
> >      STRING(REPLACE "\n" "" RUBY_CFLAGS ${RUBY_CFLAGS})
> >
> > +    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
> > +        #Clang can't handle this option (04.2014) remove this flag therefore
> > +        STRING(REPLACE "-fstack-protector-strong" "" RUBY_CFLAGS ${RUBY_CFLAGS})
>
> I guess that he is on a system where ruby got compiled using gcc and 
> he is using clang
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/rock-core/base-cmake/pull/12/files#r27040782>.
>
This is really possible ;).

Nevertheless how we handle this, in general the EABIs from clang and gcc 
are compatible. So i see no reason why we should not workaround this here.

-- 
  Dipl.-Inf. Matthias Goldhoorn
  Space and Underwater Robotic

  Universität Bremen
  FB 3 - Mathematik und Informatik
  AG Robotik
  Robert-Hooke-Straße 1
  28359 Bremen, Germany
  
  Zentrale: +49 421 178 45-6611
  
  Besuchsadresse der Nebengeschäftstelle:
  Robert-Hooke-Straße 5
  28359 Bremen, Germany
  
  Tel.:    +49 421 178 45-4193
  Empfang: +49 421 178 45-6600
  Fax:     +49 421 178 45-4150
  E-Mail:  matthias.goldhoorn@informatik.uni-bremen.de

  Weitere Informationen: http://www.informatik.uni-bremen.de/robotik